corehq/apps/app_manager/views/apps.py

```diff
@@ -574,7 +575,20 @@ def replace_all(list1, list2):
             list1.extend(list2)
     replace_all(app.langs, langs)
 
+    # Clear out old translations
+    if deleted:
+        # Get current translations, which won't include deleted languages
+        temp = raw_bulk_app_sheet(app)
+        fd, filename = tempfile.mkstemp()
+        with open(filename,'wb') as out:
+            out.write(temp.getvalue())
+
+        # Re-apply
+        with open(filename) as f:
+            msgs = process_bulk_app_translation_upload(app, f)
```

Comment:
How long do you expect this to take? I think bulk uploads are processed in celery asynchronously. If this take too long to respond to the user the request will timeout